tizen-api/src/tizen-api-pipeline.c

```diff
@@ -304,11 +368,28 @@ nns_pipeline_destroy (nns_pipeline_h pipe)
 int
 nns_pipeline_getstate (nns_pipeline_h pipe, nns_pipeline_state * state)
 {
-  /* *state = NNSAPI_UNKNOWN; */
-
-  /** @todo NYI */
+  nns_pipeline *p = pipe;
+  GstState _state;
+  GstState pending;
+  GstStateChangeReturn scret;
+  *state = NNS_PIPELINE_UNKNOWN;
+
+  if (p == NULL)
+    return NNS_ERROR_INVALID_PARAMETER;
+
+  g_mutex_lock (&p->lock);
+  scret =
+      gst_element_get_state (p->element, &_state, &pending,
+      GST_CLOCK_TIME_NONE);
+  g_mutex_unlock (&p->lock);
+
+  if (scret == GST_STATE_CHANGE_SUCCESS ||
```

Comment:
simplify to `scret != GST_STATE_CHANGE_FAILURE`

Reply:
Will be updated.